packages/algorithms/src/data_structure/lazy_segment_tree.rs

```diff
@@ -129,50 +98,46 @@ where
     }
 
     pub fn eval(&mut self, k: usize) {
-        if self.lazy[k] == (self.zero_lazy)() {
+        if self.lazy[k] == A::id() {
             return;
         }
 
-        self.data[k] = (self.mapping)(
-            self.data[k],
-            self.lazy[k],
-            k,
-            k.leading_zeros() - self.size.leading_zeros(),
```

Comment:
The previous implementation calculated not the length of the segment but the bit of length. (Actually, it's not even the bit of length. It should be `(k+1).leading_zeros() - self.size.leading_zeros()` 😢 )